### ISSUE TITLE
IE8 fails on .call

### ISSUE THREAD

**Issue by Merri** [typicode/pegasus#9]
IE8 apparently is missing onSuccess entirely and will crash. Need to add addtional check:

`if (onSuccess.call)`

->

`if (onSuccess && onSuccess.call)`

And then IE8 works. Adds 3 bytes to minified code.

**Comment by typicode** [comment]
Hey Merri,

Thanks for the issue and fix!

Do you have a failing code? Or this code fails on IE8?

```javascript
pegasus('http://jsonplaceholder.typicode.com/posts/1').then(
  function(data, xhr) {
    console.log('success');
  },
  function(data, xhr) {
    console.error('error');
  }
)
```

**Comment by Merri** [comment]
Afaik it will always fail. Can't test right now as I'm on my way home, but
the demo pages don't work for an example. Tested it earlier on Win 7 IE11
in IE8 compatibility mode and on XP, both error on the if block for
onSuccess.call. IE9+ is ok.

tiistai 24. helmikuuta 2015 typicode <notifications@github.com> kirjoitti:

> Hey Merri,
>
> Thanks for the issue and fix!
>
> Do you have a failing code? Or this code fails on IE8?
>
> pegasus('http://jsonplaceholder.typicode.com/posts/1').then(
>   function(data, xhr) {
>     console.log('success');
>   },
>   function(data, xhr) {
>     console.error('error');
>   }
> )
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/typicode/pegasus/issues/9#issuecomment-75813864>.
>


-- 
Vesa Piittinen
Kaarenjalka 5 B 19
00940 HELSINKI

**Comment by typicode** [comment]
No worries. Actually, after doing some searches on xhr requests with IE8 (can't test either right now), you must be right.

So I've added your fix and released v0.2.1, thanks again :)

**Comment by Merri** [comment]
Ok, great and thanks for the quick fix :)
